Bottom: 5e3b7a29563bb2e2cd93a7b94e3f4fa431cd9d81
Top:    1bf42e3f3bc18344fcf0c5fae724aca97fc76229
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-15 12:13:29 +0100

We can just delete kicked out holes

Since we don't keep holes from broken contours, if a contour is
kicked out, we know it is destined for the bin.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index ffb1542..f9b45f4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1390,7 +1390,8 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-	  /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
+	  /* Need to check if this new hole means we need to kick out any old
+	   * ones for deletion */
 	  while (1)
 	    {
 	      struct find_inside_info info;
@@ -1424,10 +1425,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
 	      /* Remove hole from the contour */
 	      remove_contour (pa_info->pa, prev, info.result, TRUE);
-
-	      /* Add hole as the next on the list to be processed in this very function */
-	      info.result->next = *src;
-	      *src = info.result;
+	      poly_DelContour (&info.result);
 	    }
 	  /* End check for kicked out holes */
